docs/core_docs/docs/tutorials/extraction.ipynb

```diff
@@ -1,367 +1,405 @@
+    "\n",
+    "Even though we defined our schema with the variable name `personSchema`, Zod is unable to infer this name and therefore does not pass it along to the model. To help give the LLM more clues as to what your provided schema represents, you can also give the schema you pass to `withStructuredOutput()` a name:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "id": "02f44203",
+   "metadata": {},
+   "outputs": [
     {
-      "cell_type": "markdown",
-      "id": "f07a7455-7de6-4a6f-9772-0477ef65e3dc",
-      "metadata": {},
-      "source": [
-        "## Next steps\n",
-        "\n",
-        "Now that you understand the basics of extraction with LangChain, you're ready to proceed to the rest of the how-to guides:\n",
-        "\n",
-        "- [Add Examples](/docs/how_to/extraction_examples): Learn how to use **reference examples** to improve performance.\n",
-        "- [Handle Long Text](/docs/how_to/extraction_long_text): What should you do if the text does not fit into the context window of the LLM?\n",
-        "- [Use a Parsing Approach](/docs/how_to/extraction_parse): Use a prompt based approach to extract with models that do not support **tool/function calling**."
-      ]
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "{ name: \u001b[32m'Alan Smith'\u001b[39m, hair_color: \u001b[32m'blond'\u001b[39m, height_in_meters: \u001b[32m'1.83'\u001b[39m }\n"
+     ]
     }
-  ],
-  "metadata": {
-    "kernelspec": {
-      "display_name": "Deno",
-      "language": "typescript",
-      "name": "deno"
-    },
-    "language_info": {
-      "file_extension": ".ts",
-      "mimetype": "text/x.typescript",
-      "name": "typescript",
-      "nb_converter": "script",
-      "pygments_lexer": "typescript",
-      "version": "5.3.3"
+   ],
+   "source": [
+    "const structured_llm2 = llm.withStructuredOutput(personSchema, { name: \"person\" })\n",
+    "\n",
+    "const prompt2 = await promptTemplate.invoke({\n",
+    "  text: \"Alan Smith is 6 feet tall and has blond hair.\"\n",
+    "})\n",
+    "await structured_llm2.invoke(prompt2)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "bfe7d31e",
+   "metadata": {},
+   "source": [
+    "This can improve performance in many cases."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "28c5ef0c-b8d1-4e12-bd0e-e2528de87fcc",
+   "metadata": {},
+   "source": [
+    "## Multiple Entities\n",
+    "\n",
+    "In **most cases**, you should be extracting a list of entities rather than a single entity.\n",
+    "\n",
+    "This can be easily achieved using Zod by nesting models inside one another."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "id": "591a0c16-7a17-4883-91ee-0d6d2fdb265c",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import { z } from \"zod\";\n",
+    "\n",
+    "const person = z.object({\n",
+    "  name: z.string().nullish().describe('The name of the person'),\n",
```

Comment:
I usually do `name: z.optional(z.string()).describe('The name of the person')`

I think this is fine too?